File: app/src/main/java/io/github/qwqgong/androidcyaml/Exceptions.java
package io.github.qwqgong.androidcyaml;

final class Exceptions {
    private Exceptions() {}

    static String usefulMessage(Throwable throwable) {
        String message = throwable.getMessage();
        return message == null || message.isBlank()
                ? throwable.getClass().getSimpleName()
                : message;
    }
}

File: app/src/main/java/io/github/qwqgong/androidcyaml/Exceptions.kt
package io.github.qwqgong.androidcyaml

object Exceptions {
    fun usefulMessage(throwable: Throwable): String {
        val message = throwable.message
        return if (message == null || message.isBlank()) {
            throwable.javaClass.simpleName
        } else {
            message
        }
    }
}
